### ISSUE TITLE
Patreon login with google disabled

### ISSUE THREAD

**Issue by arvenyon** [Docile-Alligator/Infinity-For-Reddit#1644]
### Checklist

- [X] I have used the search function for [open](https://github.com/Docile-Alligator/Infinity-For-Reddit/issues) **and** [closed](https://github.com/Docile-Alligator/Infinity-For-Reddit/issues?q=is%3Aissue+is%3Aclosed) issues to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.
- [X] If the bug only occurs with a certain link, post, image..., I will include the URL.

### App version

7.1.1

### Where did you get the app from

Other

### Android version

14

### Device model

Pixel 5

### Steps to reproduce

1. Download newest release atm (1.7.7) from GitHub
2. Install App
3. Get the info that you have to log in via patreon
4. "Log in with Google" button is greyed out/disabled

### Example post, link, markdown...

I figure the log-in screen for patreon is only a redirection thing used during the auth process, so not technically handled by the infinity app.
However the internal browser used to display the auth screen seems to be the issue.

The google login option being disabled when trying to login on patreon has been an issue for quite some time. Usually that happened because some necessary cookies were blocked by the browser.

### Expected behaviour

To be able to login via google

### Current behaviour

Option is disabled

**Comment by Docile-Alligator** [comment]
It's because the login page is shown inside a webview.

**Comment by IzzySoft** [comment]
@Docile-Alligator I've just been informed of the Patreon requirement, so I marked it in `NonFreeDep`. Is it true the app now also requires a paid subscription? That would make another anti-feature necessary (NonFreeAdd or NonFreeDep, or both – as the app can no longer be used "freely"). Guess that was the reason why Infinity was removed from F-Droid? Looks like I need to consider the same for IzzyOnDroid: an app that requires payment to access a non-free network is quite border-line, even if the source is under a free license.

**Comment by IzzySoft** [comment]
PS: there is no fitting anti-feature for this. And the app description is misleading here:

> It is completely free and ad-less

This contradicts the requirement of a paid subscription. What way out of this dilemma do you see, other than removing the app from IzzyOnDroid?

**Comment by Docile-Alligator** [comment]
@IzzySoft The app allows anonymous browsing for free. A subscription is only required when users login. I think I should change it to partially free?

**Comment by IzzySoft** [comment]
Yes, I agree. But then better mention that in that context. The current phrasing gives the impression it is… well, "completely free" :wink: Which is why I was reached out to.

> It is completely free and ad-less; you can browse Reddit without interruptions or distractions (a paid subscription is only required when you want to login)

or something alike, maybe?

**Comment by Docile-Alligator** [comment]
@IzzySoft It has been changed to "partially free" with updated description.

**Comment by IzzySoft** [comment]
That's putting your light under the bushel, though – unless there are other limitations when not subscribed. "partially free" could be seen as "not fully libre", which would not be true. But you can probably tell better than I what would be the best phrasing there, I can just "suggest by hearsay"…

**Comment by IzzySoft** [comment]
> the Patreon version instead of the Play version.

That's correct. I cannot find the related issue, but that was what we agreed on.

> But the paywall-less anonymous browsing mode only got implemented in the Play version, not the Patreon build.

Oof. @Docile-Alligator while I understand the background, and this is of course no show-stopper for the listing at IoD, it needs to be made clear so folks know *before* they install. I see two paths here:

* implement paywall-less anonymous browsing mode for the Patreon (and F-Droid) variants as well
* update the description accordingly (should you use Fastlane for all variants, IoD supports any starting point for Fastlane structures so we could use a different tree if needed)

> One can still compile the app yourself with your own API-key since <100 requests/min is free of charge afaik.

That's why it's no show-stopper for either F-Droid or IoD. But while F-Droid builds itself and thus could do what you're suggesting (I didn't check if it's set up there as reproducible build, which would rule out this option – or if they do patch it this way), IoD always takes the APKs from "upstream", i.e. those built & signed by the corresponding developers, and thus does not have this option.

**Comment by IzzySoft** [comment]
> You or F-Droid distributing a build with your / their own Reddit API key

Neither-or do that. F-Droid just builds from the source (yes, with some patching from time to time) but does not provide any "own" API keys. IoD, taking the "upstream APK", does not even have that option.

> All I originally intended was to point out that the updated description doesn't match the offered Patreon version

That was what I understood, yes – thanks! I was just not sure if others would understand it the same way, hence the clarification :wink:

**TL;DR:** ball is with @Docile-Alligator once more then…

**Comment by Docile-Alligator** [comment]
@IzzySoft @genericusername87 
Ah sorry for this. I totally forgot to implement a free anonymous mode for the Patreon version. This will be added very soon in the upcoming v7.2.7 update. The app description will also be changed.

**Comment by IzzySoft** [comment]
Thanks a lot, @Docile-Alligator!

**Comment by Docile-Alligator** [comment]
@IzzySoft v7.2.7 is now available.

**Comment by IzzySoft** [comment]
Oof, looks like I missed the notification. So I can close the [issue on my end](https://gitlab.com/IzzyOnDroid/repo/-/issues/547). Maybe you want to [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) to link to [Infinity at IzzyOnDroid](https://apt.izzysoft.de/packages/ml.docilealligator.infinityforreddit.patreon), @Docile-Alligator? Oh, and should the anti-feature description there need an update, please let me know.

**Comment by Docile-Alligator** [comment]
@IzzySoft Thank you! An updated badge is already in the README file! I will let you know if I need to update the anti-feature description.

**Comment by IzzySoft** [comment]
@Docile-Alligator for quite a while now (since v7.3.4), there's no `patreon` variant available anymore at your releases, so the app is no longer updated at IzzyOnDroid. Is that intended? Will it return? Or what are the next steps in this regard?

**Comment by Docile-Alligator** [comment]
@IzzySoft Hi thank you for asking. Patreon version is still supported. It just won't have beta releases. And the recent v7.4.1 update is only for solving Google Play policy issues so I don't need to release a new patreon version. BTW, infinity on Google Play has already been suspended by Google. Only Infinity+ is still available on Google Play.

**Comment by IzzySoft** [comment]
Thanks! So in other words: "nice you've noticed, but nothing to worry"? :see_no_evil: So it's just 7.4.1 not having it (7.4.0 was collapsed so I didn't see it there), and it might skip some releases. OK, then I'll sit still and let things go…

**Comment by IzzySoft** [comment]
> Patreon version is still supported and v7.4.0 was just released recently. 

Btw, built from a commit that does not exist in the repo (see embedded vcsInfo in the APK). I try to establish [Reproducible Builds](https://reproducible-builds.org/) for it since before v7.3.3 (which had the same issue), see e.g. [Reproducible Builds, special client support and more at IzzyOnDroid](https://android.izzysoft.de/articles/named/iod-rbs-mirrors-clients), but it seems I have no chance as you rarely build the APK from the commit the tag points to…

Don't mistake this for a rant please 😉 If you're interested in getting Infinity RB, you might wish to take a look at our [hints on reproducible builds](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Reproducible-Builds). Thanks!

**Comment by IzzySoft** [comment]
v7.4.4 has the same issue again: built from a commit that was never pushed.

```
fatal: Could not parse object 'e22d3079c553fcdacfa3c418fde1af1d017eb8ab'.
```

According to the embedded commit info:

```yaml
repositories {
  system: GIT
  local_root_path: "$PROJECT_DIR"
  revision: "e22d3079c553fcdacfa3c418fde1af1d017eb8ab"
}
```

Any chance for builds from a clean tree at the tagged commit that is present here, or shall I give up trying, @Docile-Alligator?

**Comment by Docile-Alligator** [comment]
@IzzySoft Sorry for the late reply. I don't think I will publish the branch for Patreon subscription so it may be best to pause any further efforts around that. Thanks for your understanding! Sorry!

**Comment by IzzySoft** [comment]
Without the code being public, is it still FOSS then? Does the license still match for this variant? Or would we have to remove the app from IzzyOnDroid entirely?

**Comment by IzzySoft** [comment]
OK, looks like we must assume the app is no longer FOSS. So as IzzyOnDroid is about FOSS apps, we'll have to remove it there now. Sorry, but I hope you understand.
